src/actions.rs

```diff
@@ -409,6 +409,75 @@ pub(crate) async fn load_policy(mut conn: &ConnectionPool) -> Result<Vec<CasbinR
     Ok(casbin_rules)
 }
 
+#[cfg(feature = "postgres")]
+pub(crate) async fn load_filtered_policy<'a>(
+    mut conn: &ConnectionPool,
+    filter: &Filter<'_>,
+) -> Result<Vec<CasbinRule>> {
+    let mut g_filter: [&str; 6] = ["%", "%", "%", "%", "%", "%"];
+    let mut p_filter: [&str; 6] = ["%", "%", "%", "%", "%", "%"];
+    for (idx, val) in filter.g.iter().enumerate() {
+        if val != &"" {
+            g_filter[idx] = val;
+        }
+    }
+    for (idx, val) in filter.p.iter().enumerate() {
+        if val != &"" {
+            p_filter[idx] = val;
+        }
```

Comment:
could you make a function for this piece of code ? So it looks better

Reply:
Extracted function in https://github.com/casbin-rs/sqlx-adapter/pull/35/commits/6360f8efe053dd353a8830e748a032c70a082d0a:

```
fn filtered_where_values<'a>(filter: &Filter<'a>) -> ([&'a str; 6], [&'a str; 6]) {
    let mut g_filter: [&'a str; 6] = ["%", "%", "%", "%", "%", "%"];
    let mut p_filter: [&'a str; 6] = ["%", "%", "%", "%", "%", "%"];
    for (idx, val) in filter.g.iter().enumerate() {
        if val != &"" {
            g_filter[idx] = val;
        }
    }
    for (idx, val) in filter.p.iter().enumerate() {
        if val != &"" {
            p_filter[idx] = val;
        }
    }
    (g_filter, p_filter)
}

```